Enterprise-CMCS/macpro-mdct-qmr
# [QMR] MDCT-2548 Run jest on PR and not commit requirement

### Description

pre-commit tests on QMR are taking forever, removing so they run on the PR per push

**prior to this change:** `./git/hooks/pre-commit` is responsible for running the script to run unit tests, and is _supposed_ to be populated by configurations set up in `pre-commit-config.yaml`. instead of `pre-commit-config.yaml` being utilized to run unit tests, they're instead being configured via a DIY pre-commit library under `.githooks/pre-commit` 

this comes into play when `./dev local` is run, specifically this command:

`# add/replace .githooks to .git/hooks
cp -a .githooks/. .git/hooks`

was causing `.git/hooks` to be populated by `.githooks/` instead of `pre-commit-config.yaml`, and hence running the custom command to run unit tests. since devs are more likely to be running `./dev local` far more than `pre-commit install`, is also the reason that prettier was likely not running correctly on most people's local dev environments 

**after this change:** since we no longer need to be running unit tests pre-commit, we can get rid of the library and the command to copy it over to `.git/hooks`, allowing `.git/hooks` to rely on `pre-commit-config.yaml` for prettier setup

### Related ticket(s)


MDCT-2548

---
### How to test

- pull changes in locally
- run `pre-commit install`
- confirm that `git commit` doesn't trigger tests and prettier works as expected (confirmed on my end)
- confirm that tests re-run on PR per push


### Important updates



- make sure to run `pre-commit install` in your local environment
- SUPER IMPORTANT: this PR will change the way files are staged r/e prettier
    - when you now attempt a commit, the pre-commit hook runs prettier
    - prettier will format the files and save them, but **will not stage them for you**
    - you can then review the files or stage them and re-commit

---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
